src/controllers/auth.controller.js

```diff
@@ -20,4 +20,28 @@ exports.signup = async (req, res) => {
     } catch (err) {
         return response_500(res, "Error creating user", err)
     }
-};
\ No newline at end of file
+};
+
+exports.login = async(req, res) =>
+{
+    const user = await User.findOne({email: req.body.email}).exec();
```

Comment:
You are querying the user base on email **3** time. In controller and twice in middleware.